src/txn/txn_rollback_to_stable.c

```diff
@@ -441,18 +441,13 @@ __rollback_abort_row_ondisk_kv(
             WT_RET(__wt_upd_alloc_tombstone(session, &upd, NULL));
             WT_STAT_CONN_INCR(session, txn_rts_keys_removed);
         }
-    } else if (vpack->tw.durable_stop_ts != WT_TS_NONE &&
+    } else if (WT_TIME_WINDOW_HAS_STOP(&vpack->tw) &&
       (vpack->tw.durable_stop_ts > rollback_timestamp || prepared)) {
         /*
          * Clear the remove operation from the key by inserting the original on-disk value as a
          * standard update.
-         *
-         * Take the value from the original page cell. If a value is simple(no compression), and is
-         * globally visible at the time of reading a page into cache, we encode its location into
-         * the WT_ROW. Otherwise, read it from the page.
          */
-        if (!__wt_row_leaf_value(page, rip, &buf))
-            WT_RET(__wt_page_cell_data_ref(session, page, vpack, &buf));
+        WT_RET(__wt_page_cell_data_ref(session, page, vpack, &buf));
```

Comment:
Could you explain this change? I don't understand why it is necessary, and then why it would fix the problem being investigated in the ticket.

Reply:
I don't understand this one as well. I made this change because I thought the check is redundant as we should never encode the value in WT_ROW if we have a stop time point. It turns out to be the magic change that stops the problem to reproduce. I had a chat with @bvpvamsikrishna and he asked me to raise this PR to test if the same set of changes also fix the problems caused by WT-6349. Unfortunately, it doesn't.

I don't believe this change fix the problem either. I think it just makes it less likely to happen.